packages/@aws-cdk/aws-rds/test/test.instance.ts

```diff
@@ -964,4 +965,112 @@ export = {
+        PolicyDocument: {
+          Statement: [{
+            Action: [
+              's3:GetObject*',
+              's3:GetBucket*',
+              's3:List*',
+            ],
+            Effect: 'Allow',
+            Resource: [
+              { 'Fn::GetAtt': ['S3ImportD5D5F2EB', 'Arn'] },
+              { 'Fn::Join': ['', [{ 'Fn::GetAtt': ['S3ImportD5D5F2EB', 'Arn'] }, '/*']] },
+            ],
+          },
+          {
+            Action: [
+              's3:GetObject*',
+              's3:GetBucket*',
+              's3:List*',
+              's3:DeleteObject*',
+              's3:PutObject*',
+              's3:Abort*',
+            ],
+            Effect: 'Allow',
+            Resource: [
+              { 'Fn::GetAtt': ['S3Export390B8694', 'Arn'] },
+              { 'Fn::Join': ['', [{ 'Fn::GetAtt': ['S3Export390B8694', 'Arn'] }, '/*']] },
+            ],
+          }],
+          Version: '2012-10-17',
+        },
+      }));
+
+      test.done();
+    },
+
+    'throws if using s3 import on unsupported engine'(test: Test) {
+      const s3ImportRole = new Role(stack, 'S3ImportRole', {
+        assumedBy: new ServicePrincipal('rds.amazonaws.com'),
+      });
+
+      test.throws(() => {
+        new rds.DatabaseInstance(stack, 'DBWithImportBucket', {
+          engine: rds.DatabaseInstanceEngine.mysql({ version: rds.MysqlEngineVersion.VER_8_0_19 }),
+          masterUsername: 'admin',
+          vpc,
+          s3ImportBuckets: [new s3.Bucket(stack, 'S3Import')],
+        });
+      }, /Engine 'mysql-8.0.19' does not support S3 import/);
+      test.throws(() => {
+        new rds.DatabaseInstance(stack, 'DBWithImportRole', {
+          engine: rds.DatabaseInstanceEngine.mysql({ version: rds.MysqlEngineVersion.VER_8_0_19 }),
+          masterUsername: 'admin',
+          vpc,
+          s3ImportRole,
+        });
+      }, /Engine 'mysql-8.0.19' does not support S3 import/);
+
+      test.done();
+    },
+
+    'throws if using s3 export on unsupported engine'(test: Test) {
+      const s3ExportRole = new Role(stack, 'S3ExportRole', {
+        assumedBy: new ServicePrincipal('rds.amazonaws.com'),
+      });
+
+      test.throws(() => {
+        new rds.DatabaseInstance(stack, 'DBWithExportBucket', {
+          engine: rds.DatabaseInstanceEngine.mysql({ version: rds.MysqlEngineVersion.VER_8_0_19 }),
+          masterUsername: 'admin',
+          vpc,
+          s3ExportBuckets: [new s3.Bucket(stack, 'S3Export')],
+        });
+      }, /Engine 'mysql-8.0.19' does not support S3 export/);
+      test.throws(() => {
+        new rds.DatabaseInstance(stack, 'DBWithExportRole', {
+          engine: rds.DatabaseInstanceEngine.mysql({ version: rds.MysqlEngineVersion.VER_8_0_19 }),
+          masterUsername: 'admin',
+          vpc,
+          s3ExportRole: s3ExportRole,
+        });
+      }, /Engine 'mysql-8.0.19' does not support S3 export/);
+
+      test.done();
+    },
+  },
```

Comment:
Are we missing a test for the Roles being different?

Reply:
Sure, good catch.